src/src/comments/comments.service.ts

```diff
@@ -0,0 +1,207 @@
+	}
+
+	async findAll(request, context, context_id, resp) {
+		const data = {
+			query: `query MyQuery {
+        comments(where: {_and: {context: {_eq: "${context}"}, context_id: {_eq: ${context_id}}}}){
+          id
+          user_id
+          comment
+          context
+          context_id
+          created_by
+          created_datetime
+          updated_by
+          updated_datetime
+          status
+          user_details{
+           first_name
+           last_name
+           id  
+          }
+        }
+      }`,
+		};
+		const response = await this.hasuraServiceFromServices.getData(data);
+		let result = response?.data?.comments;
+		if (!result || result.length < 1) {
+			return resp.status(200).send({
+				success: false,
+				status: 'Not Found',
+				message: 'Comments Not Found',
+				data: {},
+			});
+		} else {
+			return resp.status(200).json({
+				success: true,
+				message: 'Comments found success!',
+				data: result,
+			});
+		}
+	}
+
+	async findOne(id: number, context_id: any, context: any) {
+		const data = {
+			query: `query MyQuery {
+        comments(where: {_and: {id: {_eq: ${id}}, context_id: {_eq:${context_id}}}}){
+          id
+          user_id
+          comment
+          context
+          context_id
+          created_by
+          created_datetime
+          updated_by
+          updated_datetime
+          status
+          user_details{
+           first_name
+           last_name
+           id  
+          }
+          }
+      }`,
+		};
+		const result = await this.hasuraServiceFromServices.getData(data);
+		return result;
+	}
+
+	async update(
+		id: number,
+		context: any,
+		context_id: number,
+		body: any,
+		request: any,
+		response: any,
+	) {
+		const user_Id = request.mw_userid;
+		let updatedComment = {};
+		const tableName = 'comments';
+		const result = await this.findOne(id, context_id, context);
+		//if logged user and commented user is same then only able to update comment else not
+		if (
+			result.data.comments[0] &&
+			result.data.comments[0].user_id == user_Id
+		) {
+			updatedComment = await this.hasuraService.update(
+				id,
+				tableName,
+				body,
+				this.returnFields,
+				[...this.returnFields, 'id'],
+			);
+			if (updatedComment) {
+				return response.status(200).json({
+					success: true,
+					message: 'Comment Updated successfully!',
+					data: updatedComment,
+				});
+			} else {
+				return response.status(400).json({
+					success: false,
+					message: `Comment Not Updated!`,
+					data: {},
+				});
+			}
+		} else {
+			return response.status(400).json({
+				success: false,
+				message: `You Are Not Right Person To Update Comment`,
+				data: {},
+			});
+		}
+	}
+
+	async remove(
+		id: number,
+		context: any,
+		context_id: number,
+		request: any,
+		response: any,
+	) {
+		const user_Id = request.mw_userid;
+		let deleteComment = {};
+		const tableName = 'comments';
+		const commentData = await this.findOne(id, context_id, context);
```

Comment:
Remove `context` and `context_id` from function args

Reply:
resolved